src/content.ts

```diff
@@ -137,8 +137,15 @@ Array.from(document.getElementsByTagName("iframe")).forEach(map => {
                         "type": "mapResource",
                         "context": map ? serializer.serializeToString(map) : null,
                         "coordinates": [parseFloat(lat), parseFloat(lon)],
-                        "url": window.location.href,
-                        "toRender": false
+                        "toRender": "preprocess"
+                    });
+                }else if(q){ // if we don't have a lat and lon, but we have a query, send a search request
```

Comment:
Since it is else if, this block would also get executed when we have a lat long and q. Is this the expected behaviour? If not, may be you would like add extra condition to check lat long, or put the if check inside else.

Reply:
I believe this is the desired behavior since we only care about passing the q param if lat and lon aren't available as it is used to create a preprocesor API call to get the lat lon. So in the case that all three exist, only the first block should execute and we can ignore the q.